dask/dataframe/io/parquet/arrow.py

```diff
@@ -314,6 +354,99 @@ class ArrowDatasetEngine(Engine):
     # Public Class Methods
     #
 
+    @classmethod
+    def extract_filesystem(
+        cls,
+        urlpath,
+        dataset_options,
+        open_file_options,
+        storage_options,
+    ):
+
+        # Check if fs was already specified as a dataset option
+        filesystem = dataset_options.pop("filesystem", None)
+        if filesystem is None:
+            if isinstance(urlpath, (list, tuple, set)):
+                if not urlpath:
+                    raise ValueError("empty urlpath sequence")
+                strpath = stringify_path(next(iter(urlpath)))
+            else:
+                strpath = stringify_path(urlpath)
+            if (
+                strpath.startswith("s3://")
+                and not open_file_options
+                and not (
+                    # Only use PyArrow by default when storage is in s3,
+                    # and `storage_option` only includes simple options
+                    # # that are "expected" by `S3FileSystem`
+                    set(storage_options)
+                    - {
+                        "access_key",
+                        "secret_key",
+                        "session_token",
+                        "anonymous",
+                        "role_arn",
+                        "session_name",
+                        "external_id",
+                        "load_frequency",
+                        "region",
+                        "request_timeout",
+                        "connect_timeout",
+                        "scheme",
+                        "endpoint_override",
+                        "background_writes",
+                        "default_metadata",
+                        "proxy_options",
+                        "allow_bucket_creation",
+                        "allow_bucket_deletion",
+                        "retry_strategy",
+                    }
```

Comment:
We need to avoid comparing to all possible S3FileSystem arguments like this, but I'm still unsure of the best balance.

If we **always** use "arrow" as the default file-system for s3 (regardless of the contents of `storage_options`), then we will break all the s3 tests. If we bail on the "arrow" default when `storage_options` has *anything* in it, then we are requiring the user to specify `dataset={"filesystem": "arrow"}` whenever they want to configure  S3FileSystem with `storage_options`. The latter might be okay, but I'm still unsure.

Reply:
I think we should not emit the warning when falling back to `s3fs` from `pyarrow`. That should fix the test failing issue (and I think be a better user experience)